src/RestWrite.js

```diff
@@ -368,7 +371,8 @@ RestWrite.prototype.setRequiredFieldsIfNeeded = function() {
         this.data.createdAt = this.updatedAt;
 
         // Only assign new objectId if we are creating new object
-        if (!this.data.objectId) {
+        // and allow custom objectId
+        if (!this.data.objectId && !this.config.allowCustomObjectId) {
```

Comment:
Are we going to account for 0? @BufferUnderflower 
@rhuanbarreto Let me know your thoughts on this one.

Reply:
objectId will be a string, never an integer. So `"0"` is acceptable. Auto increment algorithms should be implemented on client side if they want.

Reply:
Also empty objectId should fallback to the default id generation mechanism. But again, who enables this setting must be aware of id collision probability. So also, I think that Id uniqueness validation should be implemented if it's yet not implemented.

Reply:
I don't think it should fallback because they are aware that setting `allowCustomObjectId` to `true` means that they should be the one to implement the `objectId`. Maybe we should throw an error at least in case it is empty. Uniqueness of objectId should be left on the developers who use this option.

Reply:
I agree when `objectId` key is present and empty there should be no fallback to generated id, it'll be most likely a client-side bug and throwing will make it somewhat easier to trace. To use auto generation one could simply omit the key

Reply:
Ok for me then

Reply:
Please, see my pull request on this one. :smiley: